RecoMET/METPUSubtraction/plugins/PFMETProducerMVA.cc

```diff
@@ -16,12 +16,13 @@ PFMETProducerMVA::PFMETProducerMVA(const edm::ParameterSet& cfg)
   srcPFCandidatesView_ = consumes<reco::CandidateView>(cfg.getParameter<edm::InputTag>("srcPFCandidates"));
   srcVertices_     = consumes<reco::VertexCollection>(cfg.getParameter<edm::InputTag>("srcVertices"));
   vInputTag srcLeptonsTags = cfg.getParameter<vInputTag>("srcLeptons");
+  edm::InputTag srcRho = ( cfg.exists("srcRho") ) ? cfg.getParameter<edm::InputTag>("srcRho") : (edm::InputTag("fixedGridRhoFastjetAll"));
+  consumes<double>(srcRho);
```

Comment:
You should switch to JetMETCorrections/JetCorrector/interface/JetCorrector.h
and pick up the needed JetCorrector objects from defined JetCorrectorProducer
https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/JetMETCorrections/Configuration/python/JetCorrectors_cff.py#L38